# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1350787569 -28800
# Node ID a73923b07f3f30408f8d1a8a02c8fc6347f0116e
# Parent  874feda086d21141e8d8d80568a4d1d0560843b0
add a custom title_pos to matrix_plots by default

diff --git a/sage/plot/matrix_plot.py b/sage/plot/matrix_plot.py
--- a/sage/plot/matrix_plot.py
+++ b/sage/plot/matrix_plot.py
@@ -230,13 +230,13 @@
 
 @suboptions('colorbar', orientation='vertical', format=None)
 @suboptions('subdivision',boundaries=None, style=None)
-@options(cmap='gray',marker='.',frame=True, axes=False, norm=None, 
+@options(cmap='gray',marker='.',frame=True, axes=False, norm=None,
          vmin=None, vmax=None, origin='upper',ticks_integer=True,
          subdivisions=False, colorbar=False)
 def matrix_plot(mat, **options):
     r"""
     A plot of a given matrix or 2D array.
-    
+
     If the matrix is dense, each matrix element is given a different
     color value depending on its relative size compared to the other
     elements in the matrix.  If the matrix is sparse, colors only
@@ -418,6 +418,15 @@
         sage: import numpy
         sage: matrix_plot(numpy.random.rand(10, 10))
 
+    A plot title can be added to the matrix plot.::
+
+        sage: matrix_plot(identity_matrix(50), origin='lower', title='not identity')
+
+    The title position is adjusted upwards if the ``origin`` keyword is set
+    to ``"upper"`` (this is the default).::
+
+        sage: matrix_plot(identity_matrix(50), title='identity')
+
     TESTS::
 
         sage: P.<t> = RR[]
@@ -488,6 +497,10 @@
     if options['subdivisions'] and options['subdivision_options']['boundaries'] is None:
         options['subdivision_options']['boundaries']=orig_mat.get_subdivisions()
 
+    # Custom position the title. Otherwise it overlaps with tick labels
+    if options['origin'] == 'upper' and 'title_pos' not in options:
+        options['title_pos'] = (0.5, 1.05)
+
     g = Graphics()
     g._set_extra_kwds(Graphics._extract_kwds_for_show(options))
     g.add_primitive(MatrixPlot(xy_data_array, xrange, yrange, options))
